pyupgrade.py

```diff
@@ -1196,7 +1201,8 @@ def __init__(self):  # type: () -> None
         self._class_info_stack = []  # type: List[FindPy3Plus.ClassInfo]
         self._in_comp = 0
         self.super_calls = {}  # type: Dict[Offset, ast.Call]
-        self._in_async_def = False
+        self._current_func = None  # type: Optional[SyncOrAsyncFunctionDef]
```

Comment:
this type is now overly broad here (I'd probably remove `SyncOrAsyncFunctionDef` as well)

Reply:
Would you like to change it to
```python
self._current_func = None  # type: Optional[Union[ast.AsyncFunctionDef, ast.FunctionDef, ast.Lambda]]  # noqa: E501
```

instead?

Reply:
you changed it so it can no longer contain `AsyncFunctionDef`

Reply:
Got it!

Reply:
Addressed in 56e91882b1c626d72c97882f0352d21ae93ae0fb.